core/src/net.rs

```diff
@@ -0,0 +1,64 @@
+//! Module for netlink operations
+
+use std::process::Command;
+
+use anyhow::bail;
+use nix::unistd::Pid;
+use regex::Regex;
+
+pub struct VethPair {}
+
+impl VethPair {
+    pub fn new(p1: &str, p2: &str, pid: Pid) -> anyhow::Result<Self> {
+        // Prevent CMD injections by only allowing [A-Za-z0-9_]
+        if !p1.chars().all(|c| char::is_ascii_alphanumeric(&c))
+            || !p2.chars().all(|c| char::is_ascii_alphanumeric(&c))
+        {
+            bail!("interface name is not well-formatted")
+        }
+
+        let cmd = Command::new("ip")
+            .arg("link")
+            .arg("add")
+            .arg(p1)
+            .arg("type")
+            .arg("veth")
+            .arg("peer")
+            .arg(p2)
+            .arg("netns")
+            .arg(pid.to_string())
+            .output()?;
+
+        if cmd.status.success() {
+            Ok(Self {})
```

Comment:
I really do not like that we are returning an empty struct here that serves no purpose.
With socket pairs for example, the [UnixStream::pair()](https://doc.rust-lang.org/std/os/unix/net/struct.UnixStream.html#method.pair) function returns two socket structs. 
Here we could also think about naming the struct `Veth` instead with a function `pair(p1, p2)` returning the names or two `Veth` structs containing names.

Reply:
There is a motivation behind this:

The original plan was to use the rtnetlink(7) protocol directly, instead of invoking ip(8) for the creation of the veth(4) pairs. However, as of now, this protocol is really undocumented and all crates are either too low-level or require tokio. Things may change in the future however, which is the reason, why I would like to have a type here, in order to keep the API extensible. It may need to store a kernel socket or something similar.